src/Init/Tactics.lean

```diff
@@ -607,8 +607,10 @@ syntax (name := cases) "cases " casesTarget,+ (" using " ident)? (inductionAlts)
 syntax (name := renameI) "rename_i " (colGt binderIdent)+ : tactic
 
 /--
-`repeat tac` applies `tac` to main goal. If the application succeeds,
-the tactic is applied recursively to the generated subgoals until it eventually fails.
+`repeat tac` applies `tac` to main goal.
+If the application succeeds, the tactic is applied recursively
+to the first of the generated subgoals until it eventually fails.
+Failure on one subgoal will result in `tac` not running on any later subgoals.
```

Comment:
I would suggest a more direct/iterative description, with the not-quite-recursive behavior as a caveat. Do you agree?
```suggestion
`repeat tac` repeatedly applies `tac` to the main goal until it fails.
That is, if `tac` produces multiple subgoals, only subgoals up to the first failure will be visited.
```

Reply:
it should probably also reference the `repeat'` tactic for people who want the other behavior.

Reply:
```suggestion
`repeat tac` repeatedly applies `tac` to the main goal until it fails.
That is, if `tac` produces multiple subgoals, only subgoals up to the first failure will be visited.
The `Std` library provides `repeat'` which repeats separately in each subgoal.
```